phpBB/develop/adjust_avatars.php

```diff
@@ -51,13 +51,13 @@
 	if ($new_avatar_name !== false)
 	{
 		$sql = 'UPDATE ' . USERS_TABLE . "
-			SET user_avatar = '" . $db->sql_escape($new_avatar_name) . "'
-			WHERE user_avatar = '" . $db->sql_escape($row['group_avatar']) . "' 
+			SET user_avatar = " . $db->sql_quote($new_avatar_name) . "
+			WHERE user_avatar = " . $db->sql_quote($row['group_avatar']) . "
 			AND user_avatar_type = " . AVATAR_UPLOAD;
```

Comment:
Here and in other places, phpBB core performs string escaping and then quotes it. Removing `sql_escape()` is not correct as it leads to SQL injections, so in this case it should be `$db->sql_quote($db->sql_escape)`, which doesn't look beautiful though.

Reply:
But `sql_quote()` did not exist at the time when this PR was made.
 I'm not sure where that came from, maybe the author forgot to push the necessary change as well?

Reply:
see changes of ` phpBB/phpbb/db/driver/driver.php` my function is differ than rxu function

Reply:
Strings have to be escaped, not quoted.

Reply:
really!!
How can SQL identify  it as string without quote!!???